flocx_market/common/exception.py

```diff
@@ -0,0 +1,34 @@
+class MarketplaceException(Exception):
+    msg_fmt = "An unknown exception occurred."
+    code = 500
+    safe = False
+
+    def __init__(self, message=None, **kwargs):
+        self.kwargs = kwargs
+        if 'code' not in self.kwargs:
+            try:
+                self.kwargs['code'] = self.code
+            except AttributeError:
+                pass
+
+        if not message:
+            try:
+                message = self.msg_fmt.format(**kwargs)
+            except Exception as e:
+                print(e)
+                message = self.msg_fmt
+
+        self.message = message
+        super(MarketplaceException, self).__init__(message)
+
+
+class ResourceNotFound(MarketplaceException):
+    msg_fmt = "{resource_type} {resource_uuid} not found."
+
+
+class ResourceQueryNotFound(MarketplaceException):
+    msg_fmt = "Found no {resource_type}s"
```

Comment:
This exception probably isn't necessary; querying for all of an object and getting nothing in return is legitimate.